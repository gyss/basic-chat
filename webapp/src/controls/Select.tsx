/** @jsx jsx */

import * as React from 'react'
import {css, jsx} from '@emotion/core'

const selectStyles = css`
  font-size: 1em;
`

interface Option {
  value: string
  label: string
}
interface IProps {
  options: Array<Option>
}

export default function Select({options, value, ...props}: IProps & React.InputHTMLAttributes<HTMLSelectElement>) {
  return (
    <select css={selectStyles} {...props} value={value}>
      {options.map((option: Option) => (
        <option value={option.value} key={option.value}>
          {option.label}
        </option>
      ))}
    </select>
  )
}
