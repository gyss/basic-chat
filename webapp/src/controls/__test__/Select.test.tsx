/** @format */

import * as React from 'react'
import {render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Select from '../Select'

describe('<Select>', () => {
  test('should render', () => {
    const options = [
      {value: '0', label: 'zero'},
      {value: '1', label: 'one'},
    ]
    const {asFragment} = render(<Select value="1" options={options} onChange={() => {}} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
