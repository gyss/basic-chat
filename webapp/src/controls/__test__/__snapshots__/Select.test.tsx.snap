// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select> should render 1`] = `
<DocumentFragment>
  <select
    class="css-uzlkwv-Select"
  >
    <option
      value="0"
    >
      zero
    </option>
    <option
      value="1"
    >
      one
    </option>
  </select>
</DocumentFragment>
`;
